document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -296,7 +296,9 @@ public BulkUpdateResult bulkUpdateSubDocs(Map<Key, Map<String, Document>> docume
       for (String subDocPath : subDocuments.keySet()) {
         Document subDocument = subDocuments.get(subDocPath);
         try {
-          BasicDBObject dbObject = new BasicDBObject(subDocPath, getSanitizedObject(subDocument));
+          BasicDBObject literalObject =
```

Comment:
Please add comments with the reference to the mongo bug for this workaround

Reply:
done